### PR TITLE
Validate field validations

### PR DESCRIPTION
**Pull request by zcei**

### CODE REVIEW

File: lib/migration-payloads/validation/errors.js [hunk 1]
```diff
@@ -39,6 +39,17 @@ const errors = {
     },
     NO_TYPE_CHANGE: (fieldId, ctId, parentFieldType, fieldType) => {
       return `Cannot change the type of field "${fieldId}" on content type "${ctId}" from "${parentFieldType}" to "${fieldType}". Field types cannot be changed.`;
+    },
+    validations: {
+      DUPLICATED_VALIDATION: (duplicatedValue) => {
+        return `A field can't have duplicates in the validations array. Duplicated "${JSON.stringify(duplicatedValue)}"`;
```
Comment (jfr3000) [inline lib/migration-payloads/validation/errors.js]: I'd changes this to "Duplicate: "${JSON.stringify(duplicatedValue)}"

File: lib/migration-payloads/validation/errors.js [hunk 2]
```diff
@@ -39,6 +39,17 @@ const errors = {
...
+      },
+      INVALID_VALIDATION_PROPERTY: (propName) => {
+        return `A field can't have "${propName}" as a validation.`;
+      },
+      INVALID_VALIDATION_PARAMETER: (propName, expectedType, actualType) => {
+        return `A fields' "${propName}" validation expected to be "${expectedType}", but got "${actualType}"`;
```
Comment (jfr3000) [inline lib/migration-payloads/validation/errors.js]: I'd change this to `"${propName}" validation expected "${expectedType}", but got "${actualType}"`

File: examples/09-validate-validations.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+module.exports = function (migration) {
+  const food = migration.editContentType('food');
+
+  food.createField('boo')
+    .type('Symbol')
+    .name('what it tastes like')
+    .validations([{ unique: true }]);
+
+  food.displayField('taste');
+};
```
Comment (jfr3000) [inline examples/09-validate-validations.js]: maybe we should have an integration test using this example?

Comment (zcei) [reply examples/09-validate-validations.js]: Oh, true, there's actually a ticket for that 😅

File: lib/migration-payloads/validation/errors.js (hunk 1 above)
Comment (zcei) [reply lib/migration-payloads/validation/errors.js]: But then you don't know the context of where and why there is a duplicate 🙁

Comment (jfr3000) [reply lib/migration-payloads/validation/errors.js]: Oh I only meant change "Duplicated" to "Duplicate:"

File: lib/migration-payloads/validation/field-validations.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+const assetImageDimensions = validation('assetImageDimensions', Joi.object({
+  width: range('number'),
+  height: range('number')
+}));
+
+const assetFileSize = validation('assetFileSize', range('number'));
+
+
+module.exports = Joi.alternatives().try(
```
Comment (jfr3000) [inline lib/migration-payloads/validation/field-validations.js]: nit: extra line

File: lib/migration-payloads/validation/schema-validation.js [hunk 5]
```diff
@@ -128,6 +132,28 @@ const validateContentType = function (payload) {
   });
 };
 
+const compactValidationsErrors = function (error) {
```
Comment (jfr3000) [inline lib/migration-payloads/validation/schema-validation.js]: nit: maybe something like `cleanErrors` or `cleanJoiNoiseFromErrors` would be more descriptive?

**Pull request merged**